tls/s2n_server_new_session_ticket.c

```diff
@@ -271,17 +294,17 @@ S2N_RESULT s2n_tls13_server_nst_write(struct s2n_connection *conn, struct s2n_st
     RESULT_ENSURE_REF(conn);
     RESULT_ENSURE_REF(output);
 
-    struct s2n_ticket_fields *ticket_fields = &conn->tls13_ticket_fields;
+    struct s2n_ticket_fields *ticket_fields = &conn->ticket_fields;
 
     /* Write message type because session resumption in TLS13 is a post-handshake message */
     RESULT_GUARD_POSIX(s2n_stuffer_write_uint8(output, TLS_SERVER_NEW_SESSION_TICKET));
 
     struct s2n_stuffer_reservation message_size = { 0 };
     RESULT_GUARD_POSIX(s2n_stuffer_reserve_uint24(output, &message_size));
 
-    uint32_t ticket_lifetime_in_secs = 0;
-    RESULT_GUARD(s2n_generate_ticket_lifetime(conn, &ticket_lifetime_in_secs));
-    RESULT_GUARD_POSIX(s2n_stuffer_write_uint32(output, ticket_lifetime_in_secs));
+    /* key_intro_time is not yet retrieved, so skip session ticket lifetime calculation and write it when the data is available */
+    struct s2n_stuffer_reservation ticket_lifetime_reservation = { 0 };
+    RESULT_GUARD_POSIX(s2n_stuffer_reserve_uint32(output, &ticket_lifetime_reservation));
```

Comment:
This comment needs work. My reaction was "what is key_intro_time? where is that mentioned?". What you're trying to say is that we haven't chosen a session ticket encryption key yet, so we don't know how that key will affect our ticket lifetime.

Did you consider alternatively pulling the call to s2n_get_ticket_encrypt_decrypt_key out of s2n_resume_encrypt_session_ticket so that the key is chosen earlier? That might make what you're doing more obvious.